engine/common/host.c

```diff
@@ -1070,3 +1082,26 @@ BOOL WINAPI DllMain( HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved )
 	return TRUE;
 #endif
 }
+
+#ifdef PANDORA
+void Pandora_Host_Shutdown( void )
+{
+	if( host.shutdown_issued ) return;
+	host.shutdown_issued = true;
+
+	if( host.state != HOST_ERR_FATAL ) host.state = HOST_SHUTDOWN; // prepare host to normal shutdown
+	if( !host.change_game ) Q_strncpy( host.finalmsg, "Server shutdown", sizeof( host.finalmsg ));
+
+	if( host.type == HOST_NORMAL )
+		Host_WriteConfig();
+
+	SV_Shutdown( false );
+	CL_Shutdown();
+
+	Mod_Shutdown();
+	NET_Shutdown();
+	Host_FreeCommon();
+	Con_DestroyConsole();
+
+}
+#endif

```

Comment:
Sense of new Host_Shutdown? 
If you mean bug when game catches SIGSEGV, that is already fixed in game_launch.